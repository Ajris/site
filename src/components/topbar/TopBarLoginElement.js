import React from "react";
import {Link} from "react-router-dom";

class TopBarLoginElement extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: props.name
        }
    }

    render() {
        return (
            <Link className="nav-link" to="/login">{this.state.name}</Link>
        )
    }
}

export default TopBarLoginElement;