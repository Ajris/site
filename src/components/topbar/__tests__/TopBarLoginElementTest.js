import React from "react";
import { create } from "react-test-renderer";
import TopBarLoginElement from "../TopBarLoginElement";
import {BrowserRouter} from "react-router-dom";

describe("TopBarLoginElement component", () => {
    test("it matches the snapshot", () => {
        const component = create(
            <BrowserRouter>
                <TopBarLoginElement/>
            </BrowserRouter>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});