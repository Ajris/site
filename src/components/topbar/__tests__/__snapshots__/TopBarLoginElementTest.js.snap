// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarLoginElement component it matches the snapshot 1`] = `
<div
  className="nav-item"
>
  <a
    className="nav-link top-bar-element"
    href="/login"
    onClick={[Function]}
  />
</div>
`;
